sycl/test/matrix/matrix-int8-test-use.cpp

```diff
@@ -47,53 +53,45 @@ void matrix_multiply(big_matrix<T1, NUM_ROWS_C, NUM_COLS_C> &C,
      auto accB = bufB.get_access<access::mode::read_write>(cgh);
 
      cgh.parallel_for<class imatrix>(
-         nd_range<2>({NDRangeM, NDRangeN}, {1, 1}),
-         [ accA, accB, accC, M, N, K ](nd_item<2> spmd_item)
-             [[intel::reqd_sub_group_size(1)]]
+         nd_range<2>({NDRangeM, NDRangeN * SG_SZ}, {1, 1 * SG_SZ}),
+         [accA, accB, accC, M, N, K](nd_item<2> spmd_item)
+             [[intel::reqd_sub_group_size(SG_SZ)]]
 
          {
            // The submatrix API has to be accessed by all the workitems in a
            // subgroup these functions will be called once by the subgroup no
            // code divergence between the workitems
            const auto global_idx = spmd_item.get_global_id(0);
            const auto global_idy = spmd_item.get_global_id(1);
-           const auto sg_startx = global_idx;
-           const auto sg_starty = global_idy;
+           const auto sg_startx = global_idx - spmd_item.get_local_id(0);
+           const auto sg_starty = global_idy - spmd_item.get_local_id(1);
 
            ext::oneapi::sub_group sg = spmd_item.get_sub_group();
-           joint_matrix<ext::oneapi::sub_group, int8_t, TM, TK> sub_a(sg);
+           joint_matrix<int8_t, TM, TK, matrix_use::matrix_a> sub_a(sg);
            // For B, since current implementation does not support non-packed
            // layout, users need to specify the updated VNNI sizes along with
            // the packed_b layout. By default, the layout is row_major and size
            // is (TK, TN).
-           joint_matrix<ext::oneapi::sub_group, int8_t, TK / 4, TN * 4,
-                        matrix_layout::packed_b>
-               sub_b(sg);
-           joint_matrix<ext::oneapi::sub_group, int32_t, TM, TN> sub_c(sg);
-
-           // Only the leader perform AMX computation.
-           if (spmd_item.get_local_id(1) % TILE_SZ)
-             return;
+           joint_matrix<int8_t, TK, TN, matrix_use::matrix_b> sub_b(sg);
+           joint_matrix<int32_t, TM, TN, matrix_use::accumulator> sub_c(sg);
+
            // AMX: 8 register tiles : 1k byte size, SMmaxxSKmax =16x64
            // strideX = X's cols, so strideC = N, strideA = K, strideB = N*4
-           joint_matrix_load(sg, sub_c,
-                             accC.get_pointer() + (sg_startx * TM) * N +
-                                 sg_starty * TN,
-                             N, matrix_layout::row_major);
-           for (int k = 0; k < K / TK; k += 1) { // K->int8_t
+           joint_matrix_fill(sg, sub_c, 0);
```

Comment:
I am curious about what is done if we write
```suggestion
           joint_fill(sg, sub_c, 0);
```
?